app/containers/CreateWallet/CreateWallet.jsx

```diff
@@ -15,73 +21,108 @@ type State = {
   passphrase: string,
   passphrase2: string,
   wif: string,
+  option: string,
+  walletName: string
 }
 
-export default class EncryptKey extends React.Component<Props, State> {
+export default class CreateWallet extends React.Component<Props, State> {
   state = {
     passphrase: '',
     passphrase2: '',
-    wif: ''
+    wif: '',
+    walletName: '',
+    option: 'CREATE'
   }
 
-  createWalletAccount = (e: SyntheticMouseEvent<*>) => {
+  createWalletAccount = (e: SyntheticMouseEvent<*>, options: Object) => {
     e.preventDefault()
-    const { encryptWIF, history } = this.props
-    const { passphrase, passphrase2, wif } = this.state
+    const { history } = this.props
+    const { passphrase, passphrase2, wif, walletName, option } = this.state
     const { generateNewWalletAccount } = this.props
-    if (!generateNewWalletAccount(passphrase, passphrase2, encryptWIF ? wif : null, history)) {
-      // this.resetFields()
-    }
-  }
-
-  resetFields () {
-    this.setState({
-      passphrase: '',
-      passphrase2: '',
-      wif: ''
-    })
+    generateNewWalletAccount(
+      passphrase,
+      passphrase2,
+      option === 'IMPORT' ? wif : null,
+      history,
+      walletName
+    )
   }
 
-  render () {
-    const { encryptWIF } = this.props
-    const { passphrase, passphrase2, wif } = this.state
+  render = () => {
+    const { passphrase, passphrase2, wif, option, walletName } = this.state
     let disabledButton
```

Comment:
This isn't being set anymore.  I think we could replace this local var with an `isValid` or `isDisabled` function based upon `passphrase`, `passphrase2`, etc.

Reply:
👍